server/src/player/init.rs

```diff
@@ -43,21 +46,38 @@ impl<'a> System<'a> for PlayerInitSystem {
             mut entity_types,
             mut metadata,
             level,
+            config,
         ) = data;
 
         // Run through events
         for event in events.read(&mut self.join_event_reader.as_mut().unwrap()) {
+            // Load player data
+            let uuid = event.uuid;
+            debug!("Loading player data for UUID {}", uuid);
+
+            let player_data = feather_core::player_data::load_player_data(uuid).unwrap_or_default();
+
+            // If this is a new player, set gamemode to server's default (config)
+            let default_gamemode = &config.server.default_gamemode;
+            let gamemode = match player_data.gamemode {
+                None => Gamemode::from_string(default_gamemode.to_string()),
+                Some(id) => Gamemode::from_id(id as u8),
+            };
+
             let player_comp = PlayerComponent {
                 profile_properties: event.profile_properties.clone(),
-                gamemode: Gamemode::Creative,
+                gamemode,
             };
             player_comps.insert(event.player, player_comp).unwrap();
 
-            let spawn_pos = position!(
-                f64::from(level.spawn_x),
-                f64::from(level.spawn_y),
-                f64::from(level.spawn_z)
-            );
+            let spawn_pos = match player_data.position {
+                Some(pos) => position!(pos[0], pos[1], pos[2]),
```

Comment:
This will panic if the player data file doesn't contain 3 elements in the position array. A length check should be added before this.